content/departments/product-engineering/engineering/enablement/repo-management/support-new-code-host.md

```diff
@@ -0,0 +1,52 @@
+# Adding support for a new code host
+
+Sourcegraph currently supports a number of different [code hosts](https://docs.sourcegraph.com/admin/external_service).
+
+Not all code hosts are supported "natively". What we mean by this is that the code host type is listed an an option in our UI and it seamless to add. In some cases it is possible to work around this if the code host is based on `git`, in which case it can be added via our [other git repository hosts](https://docs.sourcegraph.com/admin/external_service/other) mechanism.
+
+The following are the list of changes required to nativley support another code host type.
+
+## Steps required for all new code host types
+
+We'll use [Pagure](https://pagure.io/) as an example. Here's the [commit](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/commit/2ad7025525622f0d80da66234b51b11526da68f6) that introduced the majority of the changes required.
+
+1. Add a [feature flag](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@078c426697c5bc6fedcba5b7851f3a2c194101f6/-/blob/schema/site.schema.json?L128) to guard it until the implementation is complete
+1. [Implement](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@39265b5cf53167a86bd993a0f0081b49c18aab81/-/blob/internal/repos/pagure.go?L22) the [Source](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@39265b5/-/blob/internal/repos/sources.go?L74) interface. To do this you'll most likely need to create a [client](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@39265b5cf53167a86bd993a0f0081b49c18aab81/-/blob/internal/extsvc/pagure/client.go?L29:6) for the code host
+1. Add code host [configuration schema](https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24%4039265b5+case+*schema.PagureConnection&patternType=literal)
+1. Add a new [kind](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@39265b5cf53167a86bd993a0f0081b49c18aab81/-/blob/internal/extsvc/types.go?L73) and [type](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@39265b5cf53167a86bd993a0f0081b49c18aab81/-/blob/internal/extsvc/types.go?L91)
+1. It also needs to be added to our [GraphQL schema](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@39265b5cf53167a86bd993a0f0081b49c18aab81/-/blob/cmd/frontend/graphqlbackend/schema.graphql?L2248)
+1. Update to ensure all the cases where we check for [types](https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24%4039265b5+case+TypePagure&patternType=literal) and [kinds](https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24%4039265b5+case+KindPagure&patternType=literal) are covered
```

Comment:
Update what exactly?

Reply:
It's mostly just doing simple string mappings or returning the correct type of structure.

The surrounding code make it pretty clear what needs to be done in each case.